.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,262 @@
+          # We don't need this at all, and our
+          # builds often fails because Microsoft
+          # servers are unstable or even offline.
+          sudo rm -f /etc/apt/sources.list.d/dotnetdev.list
+          sudo rm -f /etc/apt/sources.list.d/azure*.list
+
+      - name: Install system dependencies
+        run: |
+          sudo apt-get update --quiet --yes 1>/dev/null
+          sudo apt-get install --no-install-recommends -q -y re2c
+
+      - name: Get Zephir Parser Cache Key
+        id: zephir-parser-cache-key
+        run: echo "::set-output name=key::$(echo -n ${ZEPHIR_PARSER_VERSION}_${CPUHASH})"
+
+      - name: Cache Zephir Parser
+        uses: actions/cache@v1
+        with:
+          path: ~/php-zephir-parser
+          key: ${{ runner.os }}-${{ matrix.php }}-zephir-parser-${{ steps.zephir-parser-cache.outputs.key }}
+          restore-keys: ${{ runner.os }}-${{ matrix.php }}-zephir-parser-
+
+      - name: Install Zephir Parser
+        run: |
+          if [ ! -f ~/php-zephir-parser/LICENSE ]; then
+            rm -rf ~/php-zephir-parser
+            git clone -b "$ZEPHIR_PARSER_VERSION" \
+              --depth 1 \
+              -q https://github.com/phalcon/php-zephir-parser \
+              ~/php-zephir-parser
+          fi
+
+          cd ~/php-zephir-parser
+          phpize
+          ./configure --with-php-config=/usr/bin/php-config --enable-zephir_parser
+          make -j"$(getconf _NPROCESSORS_ONLN)"
+          sudo make install
+          echo 'extension="zephir_parser.so"' | sudo tee "/etc/php/7.4/cli/conf.d/zephir_parser.ini"
+
+      - name: Setup GitHub Token
+        run: |
+          # To increase the GitHub rate limit we're use GitHub authentication
+          if [ -n "${{ secrets.GH_TOKEN }}" ]; then
+            composer config github-oauth.github.com "${{ secrets.GH_TOKEN }}"
+          fi
+
+      - name: Install Zephir
+        run: |
+          wget --no-clobber -O "$HOME/zephir" "https://github.com/phalcon/zephir/releases/download/${ZEPHIR_VERSION}/zephir.phar"
+          chmod +x "$HOME/zephir"
+      - name: Generate C code
+        run: |
+          $HOME/zephir fullclean
+          $HOME/zephir generate
+          cd build
+          php gen-build.php
+          cd ..
+
+      - name: Create pecl package
+        id: pecl_create
+        run: |
+          cp build/php7/safe/config.w32 config.w32
+          cp build/php7/safe/phalcon.zep.c phalcon.zep.c
+          cp build/php7/safe/config.m4 config.m4
+          cp build/php7/safe/php_phalcon.h php_phalcon.h
+          cp build/php7/safe/phalcon.zep.h phalcon.zep.h
+          pecl package
+          phalcon_package="`ls | grep phalcon-*tgz`"
+          mv $phalcon_package phalcon-pecl.tgz
+      - name: Validate pecl package
+        run: pecl package-validate phalcon-pecl.tgz
+      - uses: actions/upload-artifact@v1
+        with:
+          name: 'phalcon-pecl'
+          path: phalcon-pecl.tgz
+  build-and-test-with-db:
```

Comment:
could you move this to separate file?

Reply:
I need to check if I can use artefacts from a different file. Will investigate and otherwise split this.